src/utils/classifier_report.py

```diff
@@ -1,59 +1,108 @@
 from airium import Airium
-from pathlib import Path
+from collections import defaultdict
 from csv import DictReader
-import webbrowser
-import argparse
 import os
 import re
+import webbrowser
 
-
-class ReportGenerator:
-
+class DataParser:
+    def __init__(self, csv_path, url=False, substring=False):
+        self.csv = csv_path
+        self.url = url
+        self.data = False
+        self.substring = substring
+    
     # parses csv and creates a list of dictionaries  
     def parse_csv(self, csv_path):
-        
         with open(csv_path, 'r', encoding='utf-8-sig') as f:
-            self.data = list(DictReader(f))
-            
-    def normalize_urls(self, http_url, substring):
+             self.data = list(DictReader(f))
+    
+    # converts imge paths to urls based on argumet
+    def normalize_to_url(self, http_url, substring):
         for item in self.data:
             item["normalized_path"] = re.sub(rf'.+(?={substring})', http_url, item["normalized_path"])
 
+    # if mode==1 generator constructs statistics about dir tree in files
+    def create_stats(self):
+        self.stats = defaultdict(lambda: {'path': '', 'count': 0, 'count_CB': 0, 'percent_CB': 0, 'has_CB': 'False', 'avg_conf_CB': 0}) 
```

Comment:
Using a string is definitely not ideal; I used a string because the Datatables doesn't seem to know how to process the booleans when reading the data, causing the table to not show. Rather than keeping the string, `has_CB` could be a boolean within the DataParser method then converted into a string by ReportGenerator.